### ISSUE TITLE
Accessibility improvement: Role and label missing for Search, My Settings, and New Chat (Floating Action)

### ISSUE THREAD

**Issue by arielgreen** [Expensify/Expensify.cash#3101]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
In chat list, "Search", "My Settings", "New Chat(Floating Action)" should have defined role and label
## Actual Result:
In chat list, role and label are missing for "Search", "My Settings", "New Chat(Floating Action)".
## Action Performed:
1. Launch the app/webapp
2. Prerequisite: Turn ON the Screen reader.
Mobile:Navigate using Swipe gesture(From Left to Right) and verily.
Web:Navigate the page using arrow keys then verify.
## Platform:
Windows/JAWS/Chrome
MACOs/Voiceover/Safari
Android/Talkback
iOS/Voiceover
MacWebapp/Voiceover

**Notes/Photos/Videos:** 
https://user-images.githubusercontent.com/37308300/119418788-cdf71280-bcc6-11eb-91a7-c1620dac45ba.mp4

view the job [here](https://www.upwork.com/jobs/~0151b3838e3cccea19)

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jboniface (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @AndrewGable (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by TwizzyIndy** [comment]
# Proposal

1. We should add `accessibilityLabel` and `accessibilityRole` in `Header` and two `TouchableOpacity` components

https://github.com/Expensify/Expensify.cash/blob/077a793bd0871aba45d93c0db7ed7eb198cada4f/src/pages/home/sidebar/SidebarLinks.js#L133-L146

like this

```html
<Header
       textSize="large"
       title="Chats"
       accessibilityLabel="Chats"
       shouldShowEnvironmentBadge
/>
<TouchableOpacity
       accessibilityLabel="Search"
       accessibilityRole="button"
       style={[styles.flexRow, styles.ph5]}
       onPress={this.showSearchPage}
 >
<Icon src={MagnifyingGlass} />
</TouchableOpacity>
<TouchableOpacity
       accessibilityLabel="My Settings"
       accessibilityRole="button"
       onPress={this.props.onAvatarClick}
>
```

2. For `CreateMenu` component, we should also add two attributes like we did.

https://github.com/Expensify/Expensify.cash/blob/077a793bd0871aba45d93c0db7ed7eb198cada4f/src/pages/home/sidebar/SidebarScreen.js#L97-L104

# Experience
Issue #2570

**Comment by AndrewGable** [comment]
@iwiznia - Are we localizing accessibility elements? Should @TwizzyIndy's proposal include localized string logic?

**Comment by iwiznia** [comment]
Yes, I think we should definitely localize accessibility elements too.

**Comment by AndrewGable** [comment]
Ok @TwizzyIndy - Your proposal looks great, but could you add or edit your initial proposal to include the localized strings? Once I confirm that I can send the Upwork job your way.

**Comment by TwizzyIndy** [comment]
Yes, we could add the localized strings in these changes. Firstly, add some static localized strings in `en.js` and then add `withLocalizePropTypes` in prop types. So my proposed codes will be changed like:

### en.js
```js
sidebarScreen: {
        newChat: 'New Chat',
        newGroup: 'New Group',
        headerChat: 'Chats',
        buttonSearch: 'Search',
        buttonMySettings: 'My Settings',
        fabNewChat: 'New Chat(Floating Action)',
},
```

### SidebarLinks.js
```html
<Header
       textSize="large"
       title={this.props.translate('sidebarScreen.headerChat')}
       accessibilityLabel={this.props.translate('sidebarScreen.headerChat')}
       shouldShowEnvironmentBadge
/>
<TouchableOpacity
       accessibilityLabel={this.props.translate('sidebarScreen.buttonSearch')}
       accessibilityRole="button"
       style={[styles.flexRow, styles.ph5]}
       onPress={this.showSearchPage}
 >
<Icon src={MagnifyingGlass} />
</TouchableOpacity>
<TouchableOpacity
       accessibilityLabel={this.props.translate('sidebarScreen.buttonMySettings')}
       accessibilityRole="button"
       onPress={this.props.onAvatarClick}
>
```
### SidebarScreen.js
```html
<FAB
    accessibilityLabel={this.props.translate('sidebarScreen.fabNewChat')}
    accessibilityRole="menu"
    isActive={this.state.isCreateMenuActive}
    onPress={this.toggleCreateMenu}
 />
```

**Comment by AndrewGable** [comment]
Ok @TwizzyIndy thank you for your proposal, I will configure the job on the Upwork side and look forward to reviewing your pull request.

**Comment by AndrewGable** [comment]
Issue found in QA testing: https://github.com/Expensify/Expensify.cash/pull/3285#issuecomment-855161523

@TwizzyIndy - Is going to submit another PR and we will resolve the regression found.
